client/src/app/users/user.service.spec.ts

```diff
@@ -59,135 +60,237 @@ describe('User service: ', () => {
     httpTestingController.verify();
   });
 
-  it('getUsers() calls api/users', () => {
-    // Assert that the users we get from this call to getUsers()
-    // should be our set of test users. Because we're subscribing
-    // to the result of getUsers(), this won't actually get
-    // checked until the mocked HTTP request 'returns' a response.
-    // This happens when we call req.flush(testUsers) a few lines
-    // down.
-    userService.getUsers().subscribe(
-      users => expect(users).toBe(testUsers)
-    );
-
-    // Specify that (exactly) one request will be made to the specified URL.
-    const req = httpTestingController.expectOne(userService.userUrl);
-    // Check that the request made to that URL was a GET request.
-    expect(req.request.method).toEqual('GET');
-    // Check that the request had no query parameters.
-    expect(req.request.params.keys().length).toBe(0);
-    // Specify the content of the response to that request. This
-    // triggers the subscribe above, which leads to that check
-    // actually being performed.
-    req.flush(testUsers);
+  describe('When getUsers() is called with no parameters', () => {
+    it('calls `api/users`', waitForAsync(() => {
+      // Mock the `httpClient.get()` method, so that instead of making an HTTP request,
+      // it just returns our test data.
+      const mockedMethod = spyOn(httpClient, 'get').and.returnValue(of(testUsers));
+
+      // Call `userService.getUsers()` and confirm that the correct call has
+      // been made with the correct arguments.
+      //
+      // We have to `subscribe()` to the `Observable` returned by `getUsers()`.
+      // The `users` argument in the function is the array of Users returned by
+      // the call to `getUsers()`.
+      userService.getUsers().subscribe((users: User[]) => {
+        // The array of `User`s returned by `getUsers()` should be
+        // the array `testUsers`.
+        expect(users)
+          .withContext('expected users')
+          .toEqual(testUsers);
```

Comment:
These are the three lines that we _might_ want to remove if we wanted to remove all the tests that "just" check that `getUsers()` doesn't mess with the list of users returned by the `HttpClient`.

```suggestion
```

Reply:
I didn't remove it from here, but this one might also be good to remove, and the comments that are on the calls that *do* have parameters should probably be here too (I would say not instead, though, because it was more obvious in all the other places.)